src/webui/webapplication.cpp

```diff
@@ -559,19 +559,20 @@ Http::Response WebApplication::processRequest(const Http::Request &request, cons
             print(error.message(), Http::CONTENT_TYPE_TXT);
     }
 
-    header(Http::HEADER_X_XSS_PROTECTION, "1; mode=block");
-    header(Http::HEADER_X_CONTENT_TYPE_OPTIONS, "nosniff");
+    header(QLatin1String(Http::HEADER_X_XSS_PROTECTION), QLatin1String("1; mode=block"));
+    header(QLatin1String(Http::HEADER_X_CONTENT_TYPE_OPTIONS), QLatin1String("nosniff"));
 
-    QString csp = QLatin1String("default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; object-src 'none'; form-action 'self';");
-    if (m_isClickjackingProtectionEnabled) {
-        header(Http::HEADER_X_FRAME_OPTIONS, "SAMEORIGIN");
-        csp += QLatin1String(" frame-ancestors 'self';");
-    }
-    if (m_isHttpsEnabled) {
-        csp += QLatin1String(" upgrade-insecure-requests;");
-    }
+    if (m_isClickjackingProtectionEnabled)
+        header(QLatin1String(Http::HEADER_X_FRAME_OPTIONS), QLatin1String("SAMEORIGIN"));
 
-    header(Http::HEADER_CONTENT_SECURITY_POLICY, csp);
+    if (!m_isAltUIUsed) {
+        header(QLatin1String(Http::HEADER_REFERRER_POLICY), QLatin1String("same-origin"));
+
+        const QString csp = QLatin1String("default-src 'self'; style-src 'self' 'unsafe-inline'; img-src 'self' data:; script-src 'self' 'unsafe-inline'; object-src 'none'; form-action 'self';")
```

Comment:
I question if we should automatically disable this when an alt webui is enabled, considering that there's already a discrete option to enable/disable this. And other options, like `upgrade-insecure-requests`, should apply to the alt webui too.

I may consider an enhancement to allow a user to specify a custom CSP to be sent from the webserver.

Reply:
IIRC this specific CSP is tuned for built-in webUI, so it may not be suitable for others.

I'll change and serve `upgrade-insecure-requests` for both UI.

>I may consider an enhancement to allow a user to specify a custom CSP to be sent from the webserver.

I'm not really sure user would need that, IMO we can wait for a request issue before jumping right into it.

Reply:
I'm thinking we should do the same for click jacking protection. If users want to disable it, they should disable the option.